extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapLoaderTest.java

```diff
@@ -389,6 +390,54 @@ public void givenRowAndIdColumn_whenLoadAllKeys_thenReturnKeys() throws Exceptio
         assertThat(ids).contains(0);
     }
 
+    @Test
+    public void givenFalse_whenLoadAllKeys_thenReturnNull() throws Exception {
+        createTable(mapName, quote("person-id") + " INT PRIMARY KEY", "name VARCHAR(100)");
+        insertItems(mapName, 1);
+
+        Properties properties = new Properties();
+        properties.setProperty(DATA_CONNECTION_REF_PROPERTY, TEST_DATABASE_REF);
+
+        properties.setProperty(ID_COLUMN_PROPERTY, "person-id");
+        properties.setProperty(LOAD_ALL_KEYS_PROPERTY, "false");
+        mapLoader = createMapLoader(properties, hz);
+
+        List<Integer> ids = newArrayList(mapLoader.loadAllKeys());
+        assertThat(ids).isNull();
+    }
+
+    @Test
+    public void givenTrue_whenLoadAllKeys_thenReturnKeys() throws Exception {
+        createTable(mapName, quote("person-id") + " INT PRIMARY KEY", "name VARCHAR(100)");
+        insertItems(mapName, 1);
+
+        Properties properties = new Properties();
+        properties.setProperty(DATA_CONNECTION_REF_PROPERTY, TEST_DATABASE_REF);
+
+        properties.setProperty(ID_COLUMN_PROPERTY, "person-id");
+        properties.setProperty(LOAD_ALL_KEYS_PROPERTY, "true");
+        mapLoader = createMapLoader(properties, hz);
+
+        List<Integer> ids = newArrayList(mapLoader.loadAllKeys());
+        assertThat(ids).contains(0);
+    }
+
+    @Test
+    public void givenInvalid_whenLoadAllKeys_thenReturnKeys() throws Exception {
+        createTable(mapName, quote("person-id") + " INT PRIMARY KEY", "name VARCHAR(100)");
+        insertItems(mapName, 1);
+
+        Properties properties = new Properties();
+        properties.setProperty(DATA_CONNECTION_REF_PROPERTY, TEST_DATABASE_REF);
+
+        properties.setProperty(ID_COLUMN_PROPERTY, "person-id");
+        properties.setProperty(LOAD_ALL_KEYS_PROPERTY, "invalidBooleanValue");
+        mapLoader = createMapLoader(properties, hz);
```

Comment:
As per my previous comment,  this should throw an exception with a message clearly saying what was wrong with the config.

Reply:
rev: 406fc8e520aaa338af760704b54aab627c851373